### PR TITLE
fix: muting options leading to two different outcomes (AR-1818)

### PR DESCRIPTION
**Pull request by trOnk12**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1818" title="AR-1818" target="_blank">BugAR-1818</a>  [Conversation list] Clicking on "Muted" label leads to two different outcomes
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- Previous the muting option was leading to two different outcomes

### Causes (Optional)

- LaunchedEffect was not retriggered because of the key that is has is a id of the Conversation item we want to display the BottomSheet for

### Solutions

- Use inner function openConversationBottomSheet that changes the state of the ConversationBottomSheet to a Home when the same Conversation BottomSheet is retriggered with same converastion id but do not redirect to home on un mute option click

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-756/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3858509)(3858509)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3858509)(3858509)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-756/3/) **failed**.

**Review by yamilmedina — APPROVED** [review]
Great work as always master, I left open questions and some small suggestions, in case you think would make an improvement or makes sense 🚀

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationSheetState.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                ConversationSheetContent(
+                    conversationId = conversationId,
+                    title = conversationInfo.name,
+                    mutingConversationState = mutedStatus,
+                    conversationTypeDetail = ConversationTypeDetail.Private(
+                        userAvatarData.asset
+                    )
+                )
+            }
+        }
```
[the remark this answers is not in the document]
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationSheetState.kt:81]: I just tried it but the conversationInfo is not part of the base class ConversationItem() so compiler do not know how to take it from both items ;s

**Pull request merged**